### PR TITLE
chore(web): sketch layer step3

### PR DESCRIPTION
**Pull request by mkumbobeaty**
# Overview

- Refactor the initial implementation for custom properties schema

- Added custom fields based on custom properties schema  to update the feature properties of the sketch layer

### CODE REVIEW

**Review by pyshx — APPROVED** [review]
`services` and `beta/../visualizer` changes looks good to me.

File: web/src/beta/lib/core/Map/Sketch/hooks.ts [hunk 1]
```diff
@@ -236,6 +236,7 @@ export default function useHooks({
             : pluginSketchLayerFeatureAdd(selectedLayer, feature);
 
         if (layerId && featureId) {
+          console.log("callled");
```
Comment (airslice) [inline web/src/beta/lib/core/Map/Sketch/hooks.ts:239]:
```suggestion

```

File: web/src/beta/features/Editor/SketchLayerManager/index.tsx [hunk 2]
```diff
@@ -55,20 +57,26 @@ const SketchLayerManager: React.FC<SketchProps> = ({ sceneId, layerStyles, onClo
   }, []);
 
   const handleSubmit = () => {
+    const schemaJSON = propertyList.reduce((acc, property, index) => {
+      const [key] = Object.keys(property);
+      const value = `${property[key]}_${index + 1}`;
```
Comment (airslice) [inline web/src/beta/features/Editor/SketchLayerManager/index.tsx:62]: Why do we need the value to be like `Int_1`, is this requirement from backend?

Comment (airslice) [reply web/src/beta/features/Editor/SketchLayerManager/index.tsx:62]: Oh it's for the sort? Is it necessary?

File: web/src/beta/features/Editor/tabs/map/RightPanel/LayerInspector/FeatureData.tsx [hunk 3]
```diff
@@ -1,23 +1,86 @@
...
+      .sort((a, b) => {
+        const aIndex = parseInt(a.value.split("_")[1]);
+        const bIndex = parseInt(b.value.split("_")[1]);
+        return aIndex - bIndex;
+      });
+
+    const fieldArray = sortedValues.map(({ key, value }) => ({
+      id: uuidv4(),
+      type: value.replace(/_\d+$/, ""),
+      title: key.toLowerCase(),
```
Comment (airslice) [inline web/src/beta/features/Editor/tabs/map/RightPanel/LayerInspector/FeatureData.tsx:64]: Why do we need to use lower case? I think the title should present what we have in data.

File: web/src/beta/features/Editor/SketchLayerManager/index.tsx (hunk 2 above)
Comment (airslice) [reply web/src/beta/features/Editor/SketchLayerManager/index.tsx:62]: Emmm maybe it's okey, but better add a comment like `// record order for sort later`, i feel it is a little bit difficult for other developers to understand on first read.

File: web/src/beta/features/Editor/tabs/map/RightPanel/LayerInspector/CustomPropertField.tsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      value={getDynamicValue(selectedFeature, field.title, field.value)}
+      onChange={handleChange}
+    />
+  ) : field?.type === "Boolean" ? (
+    <ToggleField
+      key={field?.id}
+      name={field?.title}
+      checked={getDynamicValue(selectedFeature, field.title, field.value)}
+      onChange={handleChange}
+    />
```
Comment (airslice) [inline web/src/beta/features/Editor/tabs/map/RightPanel/LayerInspector/CustomPropertField.tsx:101]: Why some fields has `key` while the others not?

File: web/src/beta/features/Editor/SketchLayerManager/index.tsx (hunk 2 above)
Comment (mkumbobeaty) [reply web/src/beta/features/Editor/SketchLayerManager/index.tsx:62]: @airslice yes I will add the comment, this is for sorting cause when the custom properties send to backend they are ordered in alphabetic order, but what we want is user to see the custom properties in the order  he added them

File: web/src/beta/components/TabMenu/index.tsx [hunk 5]
```diff
@@ -18,11 +18,18 @@ export type TabObject = {
 export type Props = {
   tabs: TabObject[];
   selectedTab: string;
+  isScroll?: boolean;
```
Comment (airslice) [inline web/src/beta/components/TabMenu/index.tsx:21]: Could you replace all `isScroll` with `scrollable`? I think `isScroll` is a little bit confusing.

File: web/src/beta/features/Editor/SketchLayerManager/CustomedProperties/index.tsx [hunk 6]
```diff
@@ -1,107 +1,70 @@
...
-        </PropertyContentWrapper>
-      </PropertyList>
-
+          )}
+        />
+      )}
+      <AddButtonWrapper>
+        <Button icon="plus" text={t("New Property")} size="small" onClick={handlePropertyAdd} />
+      </AddButtonWrapper>
       <SubmitWrapper />
```
Comment (airslice) [inline web/src/beta/features/Editor/SketchLayerManager/CustomedProperties/index.tsx:68]:
```suggestion
```
Seems doesn't in use any more?

**Review by airslice — APPROVED** [review]

**Pull request merged**